backend/services/api-test/src/main/java/io/metersphere/api/mapper/ExtApiScenarioStepMapper.java

```diff
@@ -26,7 +26,11 @@ public interface ExtApiScenarioStepMapper {
      */
     List<String> getHasBlobRequestStepIds(@Param("scenarioId")  String scenarioId);
 
-    List<String> selectResourceId(@Param("projectId") String projectId, @Param("stepType") String stepType, @Param("protocols") List<String> apiProtocols);
+    List<String> selectResourceId(@Param("projectId") String projectId, @Param("stepType") String stepType);
+
+    List<String> selectApiResourceId(@Param("projectId") String projectId, @Param("stepType") String stepType, List<String> protocols);
+
+    List<String> selectApiCaseResourceId(@Param("projectId") String projectId, @Param("stepType") String stepType, List<String> protocols);
 
     List<String> selectCustomRequestConfigByProjectId(String projectId);
 }
```

Comment:
此代码包含两个类ExtApiScenarioStepMapper和SelectResource方法，在不同情况下有不同的功能定义，如查询资源ID或定制请求配置。根据问题的分类，我们关注的主要方面包括：

1. 类名命名一致性的维护，并且应该使用“接口”/“实现”的模式（例如：`interface ExtApiScenarioStepMapper`, `public class SelectResource`).然而在给定的示例中，没有遵循该约定。
2. 如果有多种业务逻辑需要操作同一个数据结构（List<String>），应当使用枚举或集合中的类型限定符来进行更明确的划分。

以下是一些对上述代码进行优化并改正的部分建议：
```csharp
// 定义一个Enum来表示不同的需求情况
export enum ScenarioIdRequirement { CustomRequest, CustomApiConfig }

/**
 * 获得指定ScenrioID的所有步骤引用ID列表
 *
 */
    // 正常返回方式改为return List<> instead of List<string>
    // 更好的是考虑封装成服务，将此类作为一个公共接口提供

+        // 指针方法调用来获取项目集ID
+
+        // 提供可变参数的方法，让方法可以接受多个协议
+
     /**
         * 使用枚舉來區分請求調度種類（自訂請求或自訂API配設）
     */
+        List<ScenarioIdRequirement> selectResourceFromProject(String projectId)
+        {
+            if (ScenarioIdRequirement.CustomRequest == ScenarioIdRequirement.valueOf(projectId))
```

请注意这里假设每个场景都有特定的需求类别（CustomRequest 或 CustomApiConfig）。如果有更多场景或需求，则可能还需要更多的枚举项。如果所有这些信息都属于某个共享概念，请创建另一个枚举类或者关联它们到当前类本身以增加清晰性。